src/order/views.py

```diff
@@ -157,14 +157,50 @@ def get_success_url(self):
         return self.object.get_absolute_url()
 
 
-class UpdateOrderStatus(AjaxableResponseMixin, generic.UpdateView):
-    model = Order
-    fields = ['status']
+class UpdateOrderStatus(generic.CreateView):
+    model = OrderStatusChange
+    form_class = OrderStatusChangeForm
+    template_name = "order_update_status.html"
 
     @method_decorator(login_required)
     def dispatch(self, *args, **kwargs):
         return super(UpdateOrderStatus, self).dispatch(*args, **kwargs)
 
+    def get_context_data(self, **kwargs):
+        context = super(UpdateOrderStatus, self).get_context_data(**kwargs)
+        context['order'] = get_object_or_404(
+            Order, pk=self.kwargs.get('pk')
+        )
+        context['order_status'] = ORDER_STATUS
+        context['order_no_charge'] = 'N'
+        context['order_no_charge_reasons'] = (
+            _('Restrictions mistake'),
+            _('Delivery mistake'),
+            _('Really unhappy client')
+        )
+        return context
+
+    def get_initial(self):
+        order = get_object_or_404(Order, pk=self.kwargs.get('pk'))
+        return {
+            'order': self.kwargs.get('pk'),
+            'status_from': order.status,
+            'status_to': self.request.GET.get('status'),
+        }
+
+    def form_valid(self, form):
+        response = super(UpdateOrderStatus, self).form_valid(form)
+        messages.add_message(
+            self.request, messages.SUCCESS,
+            _("The status has been changed")
+        )
+        return response
+
+    def get_success_url(self):
+        return reverse(
+            'order:view', kwargs={'pk': self.kwargs.get('pk')}
+        )
```

Comment:
The lines 193-196? In fact, I copied the same code from [member/views.py](https://github.com/savoirfairelinux/sous-chef/blob/dev/src/member/views.py#L1358-L1360). Do you mean a different message?